test/e2eTests/extension.e2e.test.ts

```diff
@@ -74,6 +83,115 @@ describe('E2E Test', () => {
         fs.writeJSONSync(detailE2EOutputFile, detailE2EOutput);
     });
 
+    it('Sign in to Docs and use real-time validation', (done) => {
+        (async function () {
+            const dispose = eventStream.subscribe(async (event: BaseEvent) => {
+                switch (event.type) {
+                    case EventType.StartLanguageServerCompleted: {
+                        if ((<StartLanguageServerCompleted>event).succeeded) {
+                            await testOpenFile();
+                            await testModifyFile();
+                            await testCreateFile();
+                            await testDeleteFile();
+
+                            dispose.unsubscribe();
+                            testEventBus.dispose();
+                            done();
+                        }
+                        break;
+                    }
+                }
+            });
+
+            triggerCommand('docs.signIn');
+
+            async function checkDiagnostics(diagnostics: { [key: string]: DiagnosticInfo[]; }) {
+                return Promise.race(
+                    [delay(timeOutMs, new TimeOutError('Timed out')),
+                    new Promise<void>(resolve => {

```

Comment:
how about extract a function which get the diagnostic with timeout and then assert it.
```

function getDiagnosticsAsync(fileUri, timeout) {
    return Promise.race(
        [delay(timeout, new TimeOutError('Timed out')),
        new Promise<Diagnostics[]>(resolve => {
            while (true) {
                await delay(1000);
                let diagnostics = vscode.languages.getDiagnostics(fileUri);
                if (diagnostics != lastDiagnostics){
                    lastDiagnostics = diagnostics
                    resolve(diagnostics);
                }
            }
        })]);
}
```